src/base/bittorrent/torrentinfo.cpp

```diff
@@ -164,8 +167,14 @@ QString TorrentInfo::name() const
 QDateTime TorrentInfo::creationDate() const
 {
     if (!isValid()) return {};
-    const boost::optional<time_t> t = m_nativeInfo->creation_date();
-    return t ? QDateTime::fromTime_t(*t) : QDateTime();
+
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    const boost::optional<time_t> date = m_nativeInfo->creation_date();
```

Comment:
@Chocobo1:  If I am reading this right, you simply removed `boost::optional` from the equation when using libtorrent 1.2.x. 

Is there any reason `boost::optional` cannot be removed from *both* sides of this `if`/`else` set?

Reply:
>Is there any reason boost::optional cannot be removed from both sides of this if/else set?

What do you mean? that type is only used in libtorrent `< 1.2.0`.